src/libexpr/value-to-json.cc

```diff
@@ -70,14 +70,17 @@ void printValueAsJSON(EvalState & state, bool strict,
             break;
         }
 
-        case tExternal:
+        case nExternal:
             v.external->printValueAsJSON(state, strict, out, context);
             break;
 
-        case tFloat:
+        case nFloat:
             out.write(v.fpoint);
             break;
 
+        case nNonNormal:
+            throw TypeError("cannot convert %1% to JSON without --strict", showType(v));
```

Comment:
Took the liberty to improve the error message slightly here. This is the only behavioral change afaik